operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/DependentResourceManager.java

```diff
@@ -90,8 +90,14 @@ private DependentResource from(DependentResourceSpec dependentResourceSpec,
       if (dependentResource instanceof KubernetesClientAware) {
         ((KubernetesClientAware) dependentResource).setKubernetesClient(client);
       }
-      dependentResourceSpec.getDependentResourceConfiguration()
-          .ifPresent(dependentResource::configureWith);
+
+      var configuration = dependentResourceSpec.getDependentResourceConfiguration();
+      if (reconciler instanceof DependentResourceConfigurationProvider) {
+        final var provider = (DependentResourceConfigurationProvider) reconciler;
+        configuration = provider.configurationFor(dependentResourceSpec);
```

Comment:
we should just pass the config part here not the whole spec.

Reply:
Not sure I understand… The whole idea is to provide the configuration so if we're passing it, why are we providing it? 😄

Reply:
Ahh sorry I miss read it / did not check it properly. Hmm don't think the Reconciler should be a provider.

Reply:
let's take a look on this togther.